src/ops/shell.rs

```diff
@@ -0,0 +1,87 @@
+//! Open up a project shell
+
+use crate::builder;
+use crate::builder::RunStatus;
+use crate::nix::CallOpts;
+use crate::ops::error::{ExitError, OpResult};
+use crate::project::{roots::Roots, Project};
+use crossbeam_channel as chan;
+use slog_scope::debug;
+use std::io;
+use std::io::Write;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+use std::time::{Duration, Instant};
+use std::{env, fs, thread};
+
+/// See the documentation for lorri::cli::Command::Shell for more
+/// details.
+pub fn main(project: Project) -> OpResult {
+    let shell = env::var("SHELL").expect("lorri shell requires $SHELL to be set");
```

Comment:
Can we handle non-POSIX `SHELL` like `fish`?

Reply:
I've tested this with `fish` and `elvish`. One issue is that sub-shells appear not to overwrite `$SHELL`: my default shell is bash, and when I create a fish sub-shell, `$SHELL` inside the fish shell still points to the bash binary. But my assumption is that when fish is your default shell, `$SHELL` points to the fish binary and will thus get picked up by `lorri shell`.

To test whether `lorri shell` works with fish and elvish in principle, I manually set `$SHELL` to their respective binary paths and then launched `lorri shell`, which successfully launched fish and elvish.

Reply:
Feel free to try this yourself with fish! :)